7 flyway

Tefaier opened the pull request:
7-ое задание
Все в основном такое же. Только для наслоения с прошлым кодом для репозитория на основе локальной памяти немного громоздко получилось. Особенно с запросами статей, потому что в map они хранились сразу же с их комментариями. Запрос же на основе дб их собирает и добавляет в статью.
Для нормальной работы с изменением статьи добавил метод edit (в ArticleRepository), который сам отвечает за внесение изменений. Метод replace в бд кидает исключение, потому что все что должно быть есть в edit, а удаление комментариев через commentrepository и не надо согласовывать, если в виде бд работа.
У меня еще нет запуска приложения на основе параметров из config (в тестах есть, но там оно берет параметры их контейнера). Поэтому, чтобы показать его использования, в Article условие trending подгружается оттуда.
Два теста ArticleControllerTest(DB) почти один в один. В общем мог вынести проверку в default, как сделал с репозиториями, но лично хочу просто удалить старый на основе map, потому что он уже бесполезен (есть отдельный тест для репозиториев).

Commit:
initial setup

Commit:
EVERYTHING!

BMSTU732 commented:
Давайте все же сделаем чтобы flyway использовался для создания нужных вам таблиц при старте приложения, а данные для подключения к БД тянулись из конфига)

Commit:
adjustments:
filled properties
made main launch file that uses docker (not temporary container)

Commit:
added some tests
excluded Main from coverRatio test

Tefaier commented:
-_-
У меня flyway и так использовался тестами, чтобы заполнить структуру в пустом контейнере.
Добавил Main.class, в котором запускается приложение используя DB варианты репозиториев, flyway, docker контейнер. Проверил немного с помощью postman'а, оно нормально подключилось к дб и развернуло сайт.
Также заметил, что у меня не было теста на addArticles -> добавил.
Main.class я кстати исключил из проверки, потому что из-за него падает проверка на покрытие, а запустить его в тесте я не могу, потому что оно пытается подключиться в бд по информации из config.
